components/brave_wallet_ui/page/async/wallet_page_async_handler.ts

```diff
@@ -60,7 +97,9 @@ handler.on(WalletPageActions.restoreWallet.getType(), async (store: Store, paylo
   keyringService.notifyWalletBackupComplete()
   await refreshWalletInfo(store)
   store.dispatch(WalletPageActions.setShowIsRestoring(false))
-  store.dispatch(WalletPageActions.walletSetupComplete())
+  if (payload?.completeWalletSetup) {
+    store.dispatch(WalletPageActions.walletSetupComplete(payload.completeWalletSetup))
+  }
```

Comment:
We now have the option to create wallets without also exiting the onboarding flow